### PR TITLE
Refactor bl_crypto_client (and bl_crypto)

### PR DESCRIPTION
**Pull request by oyvindronningstad**
Trying to disentangle some bits in the bootloader code since it's a bit hard to work with. This should hopefully also make it easier to use the client implementation of bl_crypto.

Related to #1223

### CODE REVIEW

File: subsys/bootloader/Kconfig [hunk 1]
```diff
@@ -165,9 +154,8 @@ config SB_BPROT_IN_DEBUG
 endif # SECURE_BOOT_DEBUG
 endif # IS_SECURE_BOOTLOADER
 
-if IS_SECURE_BOOTLOADER || SECURE_BOOT || SECURE_BOOT_CRYPTO_CLIENT
```
Comment (sigvartmh) [inline subsys/bootloader/Kconfig]: Is this correct has dependencies be changed?

File: samples/bootloader/prj.conf [hunk 2]
```diff
@@ -15,15 +15,18 @@ CONFIG_SYS_CLOCK_EXISTS=n
 CONFIG_NRF_RTC_TIMER=n
 CONFIG_CLOCK_CONTROL=n
 CONFIG_FPROTECT=y
+CONFIG_FW_INFO=y
```
Comment (sigvartmh) [inline samples/bootloader/prj.conf]: Isn't `FW_INFO` selected by `SECURE_BOOT_CRYPTO` ?

File: tests/subsys/bootloader/bl_crypto/prj.conf [hunk 3]
```diff
@@ -10,4 +10,8 @@ CONFIG_TEST_USERSPACE=n
 CONFIG_USERSPACE=n
 CONFIG_STDOUT_CONSOLE=n
 CONFIG_SECURE_BOOT=y
```
Comment (sigvartmh) [inline tests/subsys/bootloader/bl_crypto/prj.conf]: FW_INFO selected by SECURE_BOOT?

File: subsys/bootloader/Kconfig (hunk 1 above)
Comment (sigvartmh) [reply subsys/bootloader/Kconfig]: I think I saw changes that allowed this to be the case but I'm not sure so just an answer her would suffice.

Comment (oyvindronningstad) [reply subsys/bootloader/Kconfig]: Yes, it's correct. The _CRYPTO_CLIENT config is gone completely, SECURE_BOOT_CRYPTO is used instead. But the include of bl_crypto is gone from inside this `if` anyways :).

**Comment by oyvindronningstad** [comment]
> Address the issues with FW_INFO with either `depends on` or don't select it twice.

Thanks @sigvartmh I changed some `select`s to `depends on`

**Pull request merged**